sdk/common/src/main/java/io/opentelemetry/sdk/common/InstrumentationLibraryInfo.java

```diff
@@ -48,6 +45,7 @@ public static InstrumentationLibraryInfo empty() {
    *
    * @return the name of the instrumentation library.
    */
+  @Nullable
```

Comment:
Hmm - going from non-null to Nullable isn't technically a safe change, as can be seen with the amount of null checks we had to add everywhere (any custom exporter out there would have the same problem on a minor version bump). Since the spec says SHOULD preserve original value, but we know preserving null in Java is generally a bad idea, can we just skip this change?

Reply:
It's a fine question. @bogdandrutu was the one who pushed for this in the specs. Opinions?

Honestly, I also think this is a bad change. I would much prefer to have a fallback name, so that exporters don't all have to guard against this case. 

Can we maybe use empty string as the default, rather than keeping nulls?

Reply:
Empty string is still a better default than keeping nulls indeed.

Reply:
I'd definitely vote against setting this nullable as the value will propagate, causing lots of extra null-checks (as documented in this PR).

Reply:
Please your opinions to https://github.com/open-telemetry/opentelemetry-specification/issues/1619  :)

Reply:
Added :)